src/main/java/com/conveyal/r5/streets/StreetLayer.java

```diff
@@ -209,6 +211,16 @@ public StreetLayer() {
         speedLabeler = new SpeedLabeler(SpeedConfig.defaultConfig());
     }
 
+    public StreetLayer(TransportNetworkConfig config) {
+        this();
+        if (config != null && config.traversalPermissionLabeler != null) {
+            // TODO cleaner mapping
```

Comment:
This can be done with switch expressions, for example:

```
        if (config != null) {
            permissionLabeler = switch (config.traversalPermissionLabeler) {
                case "sidewalk" -> new SidewalkTraversalPermissionLabeler();
                case null -> new USTraversalPermissionLabeler();
                default -> throw new IllegalArgumentException("Unknown traversal permission labeler: " + config.traversalPermissionLabeler);
            };
        } else {
            permissionLabeler = new USTraversalPermissionLabeler();
        }
```